tools/verify-terms.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+# Copyright 2017 The authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script will scan all md (markdown) files for bad references.
+# It will look for strings of the form [...](...) and make sure that
+# the (...) points to either a valid file in the source tree or, in the
+# case of it being an http url, it'll make sure we don't get a 404.
+#
+# Usage: verify-links.sh [ dir | file ... ]
+# default arg is root of our source tree
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$( cd $(dirname "${BASH_SOURCE}")/.. && pwd)
+
+verbose=""
+debug=""
+stop=""
+
+while [[ "$#" != "0" && "$1" == "-"* ]]; do
+  opts="${1:1}"
+  while [[ "$opts" != "" ]]; do
+    case "${opts:0:1}" in
+      v) verbose="1" ;;
+      d) debug="1" ; verbose="1" ;;
+      -) stop="1" ;;
+      ?) echo "Usage: $0 [OPTION]... [DIR|FILE]..."
+         echo "Verify all terms defined in spec are cased correctly."
+         echo
+         echo "  -v   show each file as it is checked"
+         echo "  -?   show this help text"
+         echo "  --   treat remainder of args as dir/files"
+         exit 0 ;;
+      *) echo "Unknown option '${opts:0:1}'"
+         exit 1 ;;
+    esac
+    opts="${opts:1}"
+  done
+  shift
+  if [[ "$stop" == "1" ]]; then
+    break
+  fi
+done
+
+# echo verbose:$verbose
+# echo debug:$debug
+# echo args:$*
+
+function contains() {
+  rc=0
+  echo "$2" | grep -qi "\([[:space:]]\|^\)$3\([[:space:]]\|$\)" || return 0
+  echo "$2" | grep -q "\([[:space:]]\|^\)$3\([[:space:]]\|$\)" || {
+    echo $file - $1: Use \'$3\'
+	rc=1
+  }
+  return $rc
+}
+
+arg=""
+
+if [ "$*" == "" ]; then
+  arg="${REPO_ROOT}"
+fi
+
+Files=$(find -L $* $arg \( -name "*.md" -o -name "*.htm*" \) | sort)
+
+rc=0
+
+for file in ${Files}; do
+  # echo scanning $file
+  dir=$(dirname $file)
+
+  [[ -n "$verbose" ]] && echo "> $file"
+
+  lineNum=0
+  cat ${file} | while read line; do 
+    ((lineNum++)) || true
+
+	for rfc in "Service Binding"; do
```

Comment:
Would it make sense to also check for the other terms like "Service Instance", etc... ?

Reply:
+1 to this. It may also be better to add the tool to a separate commit once the terms are fixed on master, so that the build will always pass on master (if we add service instance to this without fixing master, it would fail)

Reply:
This is true. But it would cause failures at the moment. Here is the plan:

I am going to work through all the terms defined and Caps them. And in that PR add the term to this list. But I am going to do them one term at a time. Then future PRs will not be able to add uncapped terms!

Reply:
If we need to I can break out the script in another PR. I felt like it was easy enough to understand together. Please advise.